### PR TITLE
Documenting new installer option --foreman-ipa-authentication

### PR DESCRIPTION
**Pull request by pkamenickova**
This is a documentation for https://github.com/theforeman/puppet-foreman/pull/199

### CODE REVIEW

File: _includes/manuals/1.6/5.7.1_quick_configuration.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Then we install Foreman.
+{% highlight bash %}
+foreman-installer --foreman-ipa-authentication=true
+{% endhighlight %}
+This option can be used for the reconfiguration of existing installation as well.
+
+Some of required packages may not be available in older RHEL versions. You can get them from Jan Pazdziora's copr yum repo at http://coprfe.cloud.fedoraproject.org/coprs/adelton/identity_demo/ . Just choose the correct `.repo` file:
+{% highlight bash %}
+foreman-installer --foreman-ipa-authentication=true --foreman-configure-ipa-repo=true <.repo file URL>
```
Comment (ares) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: do we have to select the repo file? I though enabling ipa repo would be enough

Comment (adelton) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: Nod -- the URL part is not correct: https://github.com/pkamenickova/theforeman.org/pull/2

**Comment by pkamenickova** [comment]
You are right. I've fixed it in 29f7667

File: _includes/manuals/1.6/5.7.1_quick_configuration.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ipa service-add HTTP/<the-foreman-fqdn>
+{% endhighlight %}
+
+Then we install Foreman.
+{% highlight bash %}
+foreman-installer --foreman-ipa-authentication=true
+{% endhighlight %}
+This option can be used for the reconfiguration of existing installation as well.
+
+Some of required packages may not be available in older RHEL versions. You can get them from Jan Pazdziora's copr yum repo at http://coprfe.cloud.fedoraproject.org/coprs/adelton/identity_demo/ .
```
Comment (ares) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: DNS resolv issue? Is the address correct?

EDIT:
seems like you're missing - in copr-fe

File: _includes/manuals/1.6/5.7_external_authentication.md [hunk 3]
```diff
@@ -1,3 +1,2 @@
 
-The following tutorial explains how to set up Foreman authentication against FreeIPA (or Identity Management) server. We will use an external FreeIPA server and enable single sign-on based on Kerberos-tickets using **mod_auth_kerb**, access control with **mod_authnz_pam**, PAM (pluggable authentication module) authentication via **mod_intercept_form_submit**, and automatic population of Foreman users and their attributes with the help of **mod_lookup_identity** and **sssd-dbus**.
-
+The following tutorial explains how to set up Foreman authentication against FreeIPA (or Identity Management) server. First part of the tutorial describes how to configure Foreman machine via Foreman installer options. The second one shows how to achieved the same result without using these options.
```
Comment (ares) [inline _includes/manuals/1.6/5.7_external_authentication.md]: nitpick s/achieved/achieve/

File: _includes/manuals/1.6/5.7.1_quick_configuration.md (hunk 2 above)
Comment (adelton) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: Good catch. It needs to be either
http://copr-fe.cloud.fedoraproject.org/coprs/adelton/identity_demo/
or
http://copr.fedoraproject.org/coprs/adelton/identity_demo/

Comment (pkamenickova) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: Fixed in d337a34

File: _includes/manuals/1.6/5.7_external_authentication.md (hunk 3 above)
Comment (pkamenickova) [inline _includes/manuals/1.6/5.7_external_authentication.md]: Thanks! Fixed in d337a34

**Comment by ares** [comment]
Just minor thing, you use i386 repositories as examples, does it make sense to use rather x86_64? Or maybe mention both? (http://copr-fe.cloud.fedoraproject.org/coprs/adelton/identity_demo/repo/epel-6-x86_64/)
Otherwise looks good.

**Comment by adelton** [comment]
The both contain the same content, using $basearch there.

The latest copr offers
https://copr.fedoraproject.org/coprs/adelton/identity_demo/repo/epel-6/adelton-identity_demo-epel-6.repo

as non-arch-specific URL so maybe we could go with this one.

**Comment by pkamenickova** [comment]
Changed in 2236a28

**Comment by ares** [comment]
LGTM :+1:

File: manuals/1.6/index.md [hunk 4]
```diff
@@ -208,16 +208,18 @@ version: 1.6
...
-### 5.7.3 PAM Authentication
-{%include manuals/1.6/5.7.3_pam_authentication.md %}
-### 5.7.4 Populate users and attributes
-{%include manuals/1.6/5.7.4_populate_users_attributes_groups.md %}
-### 5.7.5 Namespace separation
-{%include manuals/1.6/5.7.5_namespace_separation.md %}
+### 5.7.1 Configuration via Foreman installer
+{%include manuals/1.6/5.7.1_quick_configuration.md %}
+### 5.7.2 HBAC configuration
+{%include manuals/1.6/5.7.2_HBAC_rules.md %}
```
Comment (domcleal) [inline manuals/1.6/index.md:214]: This doesn't render, the file is missing a .md extension.

(Test the site using `rake server`)

File: _includes/manuals/1.6/5.7.2_HBAC_rules [hunk 5]
```diff
@@ -0,0 +1,51 @@
+
+We suppose that the Foreman machine is FreeIPA-enrolled and *HTTP/<the-foreman-fqdn>* service has been created on FreeIPA server.
```
Comment (domcleal) [inline _includes/manuals/1.6/5.7.2_HBAC_rules]: `&lt;` and `&gt;` here so it's not interpreted as HTML, or use backticks

File: _includes/manuals/1.6/5.7.2_HBAC_rules [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+At first we create HBAC (host-based access control) service and rule on the FreeIPA server. In the following examples, we will use PAM service name *foreman-prod*.
```
Comment (domcleal) [inline _includes/manuals/1.6/5.7.2_HBAC_rules]: "we will use the PAM service name"

File: _includes/manuals/1.6/5.7.1_quick_configuration.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+This option can be used for the reconfiguration of existing installation as well.
+
+Some of required packages may not be available in older RHEL versions. You can get them from Jan Pazdziora's copr yum repo at http://copr-fe.cloud.fedoraproject.org/coprs/adelton/identity_demo/ .
+{% highlight bash %}
+foreman-installer --foreman-ipa-authentication=true --foreman-configure-ipa-repo=true
+{% endhighlight %}
+
+In case you want to use IPA server's host-based access control (HBAC) features (make sure *allow_all* rule is disabled), the default PAM service name which would be matched by HBAC service name is *foreman*, and you can override the default with:
+{% highlight bash %}
+foreman-installer --foreman-ipa-authentication=true --pam-service=<pam_service_name>
```
Comment (domcleal) [inline _includes/manuals/1.6/5.7.1_quick_configuration.md]: `--foreman-pam-service`

File: _includes/manuals/1.6/5.7.2_HBAC_rules [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+{% highlight bash %}
+ipa hbacrule-find foreman-prod
+ipa hbactest --user=<username> --host=<the-foreman-fqdn> --service=foreman-prod
+{% endhighlight %}
+
+Chances are there will be HBAC rule *allow_all* matching besides our new *allow_foreman_prod* rule. See http://www.freeipa.org/page/Howto/HBAC_and_allow_all for steps to disable the catchall *allow_all* HBAC rule while maintaining the correct operation of your FreeIPA server and enrolled clients. The goal is only *allow_foreman_prod* matching when checked with `ipa hbactest`.
+
+Next step is to define matching PAM service on the Foreman machine. We create file `/etc/pam.d/foreman-prod` with the following content:
+
+{% highlight bash %}
```
Comment (domcleal) [inline _includes/manuals/1.6/5.7.2_HBAC_rules]: Isn't this what foreman-installer does when you supply the PAM service name?  It's not clear to me if this section replaces or supplements the installer section above.

**Comment by pkamenickova** [comment]
Dominic, thanks for comments. I've fixed the issues (and copr URL inconsistency) in 8c9b5cd.

**Comment by domcleal** [comment]
Thanks very much @pkamenickova!

**Pull request merged**